x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/job/config/Job.java

```diff
@@ -1344,11 +1367,26 @@ public Job build() {
 
             if (Strings.isNullOrEmpty(resultsIndexName)) {
                 resultsIndexName = AnomalyDetectorsIndexFields.RESULTS_INDEX_DEFAULT;
-            } else if (resultsIndexName.equals(AnomalyDetectorsIndexFields.RESULTS_INDEX_DEFAULT) == false) {
-                // User-defined names are prepended with "custom"
-                // Conditional guards against multiple prepending due to updates instead of first creation
-                resultsIndexName = resultsIndexName.startsWith("custom-") ? resultsIndexName : "custom-" + resultsIndexName;
+            } else if ((resultsIndexName.startsWith(AnomalyDetectorsIndexFields.RESULTS_INDEX_SHARED)
+                && MlIndexAndAlias.has6DigitSuffix(resultsIndexName)
+                && resultsIndexName.length() == AnomalyDetectorsIndexFields.RESULTS_INDEX_DEFAULT.length()) == false) {
```

Comment:
This condition is difficult to grasp. Maybe you can simplify it or extract a method with a meaningful name.